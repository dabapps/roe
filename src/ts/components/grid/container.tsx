import * as classNames from 'classnames';
import * as React from 'react';
import { HTMLProps, PureComponent } from 'react';
import { ComponentProps } from '../../types';

export interface ContainerProps extends ComponentProps, HTMLProps<HTMLElement> {
  /**
   * Fill parent with no media queries to affect width.
   */
  fluid?: boolean;
  /**
   * Allows applying a background color with `@container-background` variable.
   */
  solid?: boolean;
}

/**
 * Used inside `NavBar`s or as the main wrapper for an application.
 */
export class Container extends PureComponent<ContainerProps, {}> {
  public render() {
    const {
      children,
      className,
      fluid,
      solid,
      component: Component = 'div',
      ...remainingProps
    } = this.props;
    const fluidClassName = fluid ? 'container-fluid' : 'container';
    const solidClassName = solid && 'solid';

    return (
      <Component
        {...remainingProps}
        className={classNames(fluidClassName, solidClassName, className)}
      >
        {children}
      </Component>
    );
  }
}

export default Container;
