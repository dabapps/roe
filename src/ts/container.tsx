import * as classNames from 'classnames';
import * as React from 'react';

export interface IProps {
  fluid?: boolean;
  solid?: boolean;
  children?: React.ReactNode;
}

export const Container: React.SFC<IProps & React.HTMLProps<HTMLDivElement>> = (props) => {
  const { children, className, fluid, solid, ...remainingProps } = props;
  const fluidClassName = fluid ? 'container-fluid' : 'container';
  const solidClassName = solid && 'solid';

  return (
    <div {...remainingProps} className={classNames(fluidClassName, solidClassName, className)}>
      {children}
    </div>
  );
};
