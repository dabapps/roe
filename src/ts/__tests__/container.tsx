import * as React from 'react';
import * as renderer from 'react-test-renderer';

import { Container } from '../container';

describe('Container', () => {

  it('should match snapshot', () => {
    const tree = renderer.create(
      <Container />
    );

    expect(tree).toMatchSnapshot();
  });

  it('should take an optional fluid prop', () => {
    const tree = renderer.create(
      <Container fluid />
    );

    expect(tree).toMatchSnapshot();
  });

});
