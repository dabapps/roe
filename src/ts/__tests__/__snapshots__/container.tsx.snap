// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container should match snapshot 1`] = `
<div
  className="container"
/>
`;

exports[`Container should take an optional fluid prop 1`] = `
<div
  className="container-fluid"
/>
`;
